### PR TITLE
wazevo(frontend): BCE should not propagate the memory address to unsealed blocks

### PR DESCRIPTION
**Pull request by evacchi**
tl;dr: I propose to propagate the memory address only for **sealed** blocks, where we know that no predecessors will be added at a later time.

This originated from an issue raised by @jerbob92 for go-pdfium.

Essentially, it looks like block 4 of function `5011` (local ID = 4960) in go-pdfium references some values that *may* not be visibile on by some of its predecessors.

We are propagating known bounds _while_ we are creating basic blocks; which means that, at a give time, we _might not know_ all of the possible predecessors to a given block. In the case of that block 4, the SSA dump shows:

```
blk4: (v305:i32,v341:i32) <-- (blk3,blk62)
```

which means that one of the predecessors in `blk62`. However, when we enter function `initializeCurrentBlockKnownBounds()` we enter `case 1` because there is only one predecessor (`blk3`) at that time. 

When blocks have more than 1 predecessor, we should enter the `default` case instead: this never propagates `kb.bound` AND `kb.address`, but it always propagates `kb.bound` and `ssa.ValueInvalid` because it cannot assume that the address will be valid on all paths.

This small change, as opposed to other earlier attempts, where I was unconditionally propagating `ssa.ValueInvalid` in all cases (see the Slack threads), does not affect codegen in a lot of cases (e.g. test case `TestCompiler_LowerToSSA/bounds check in if-else` is left unaffected)

### CODE REVIEW

**Comment by mathetake** [comment]
this is exactly what I was thinking is the right way to fix the issue, so looks good

**Comment by evacchi** [comment]
Zig Stdlib (arm64)
```
main: 45877240
this: 45877128
```

Zig Stdlib (amd64)
```
main: 44953787
this: 44954715
```

apparently this is actually slightly _better_ on arm64... (I checked multiple times)

**Review by mathetake — APPROVED** [review]

**Comment by mathetake** [comment]
once https://github.com/tetratelabs/wazero/issues/2158 is resolved as well, let's run the fuzzing for both arm64 and amd64 for like multiple days (let's not forget to add `--no-trace-compares` flag: https://github.com/tetratelabs/wazero/pull/2152)
